qualang_tools/plot/Fitting/fitting.py

```diff
@@ -0,0 +1,418 @@
+        - Fitting to transmission resonator spectroscopy
+        - Fitting to reflection resonator spectroscopy
+
+    """
+
+    @staticmethod
+    def linear(x, y):
+        """
+           Create a linear fit of the form
+
+           .. math::
+           f(x) = a * x + b
+
+           for unknown parameters :
+                a - The slope of the function
+                b - The free parameter of the function
+
+
+           :param x: The data on the x axis
+           :param y: The data on the y axis
+           :return: A dictionary of (fit_func, a, b)
+
+           """
+
+        # Normalizing the vectors
+        xn = preprocessing.normalize([x], return_norm=True)
+        yn = preprocessing.normalize([y], return_norm=True)
+        x = xn[0][0]
+        y = yn[0][0]
+
+        # Finding an initial guess to the slope
+        m0 = (y[-1] - y[0]) / (x[-1] - x[0])
+
+        # Finding an initial guess to the free parameter
+        n0 = y[0]
+
+        fit_type = lambda x, a: m0 * a[0] * x + n0 * a[1]
+
+        def curve_fit(f, x, y, a0):
+            def opt(x, y, a):
+                return np.sum(np.abs(f(x, a) - y) ** 2)
+
+            out = optimize.minimize(lambda a: opt(x, y, a), a0)
+            if not out.success: raise Exception(
+                "The fitting failed, please check the quality of the data or the fitting function that you are using")
+            return out["x"]
+
+        popt = curve_fit(fit_type, x, y, [m0, n0])
+
+        print(
+            f"a = {popt[0] * yn[1] / xn[1]}, b = {popt[1] * yn[1]}"
+        )
+        out = {
+            "fit_func": lambda x: fit_type(x, popt) * yn[1],
+            "a": popt[0] * yn[1] / xn[1],
+            "b": popt[1] * yn[1],
+        }
+
+        return out
+
+    @staticmethod
+    def ramsey(x, y):
+        """
+          Create a fit to Ramsey experiment of the form
+
+          .. math::
+          f(x) = uncertainty_population * (1 - np.exp(-x * (1/T2))) + amp / 2 * (
+              np.exp(-x * (1/tau))
+              * (initial_offset * 2 + np.cos(2 * np.pi * f * x + phase))
+              )
+
+          for unknown parameters :
+              f - The detuning frequency
+              phase - The phase
+              T2 - The decay constant
+              amp - The amplitude
+              uncertainty_population -  The uncertainty in population
+              initial_offset - The initial offset
+
+          :param x:  The dephasing time in ns
+          :param y: The transition probability
+          :return: A dictionary of (fit_func, f, phase, tau, amp, uncertainty_population, initial_offset)
+
+           """
+
+        # Normalizing the vectors
+        xn = preprocessing.normalize([x], return_norm=True)
+        yn = preprocessing.normalize([y], return_norm=True)
+        x = xn[0][0]
+        y = yn[0][0]
+
+
+        w = np.fft.fft(y)
+        freqs = np.fft.fftfreq(len(x))
+        new_w = w[1: len(freqs // 2)]
+        new_f = freqs[1: len(freqs // 2)]
```

Comment:
I guess that it should be `w[1: len(freqs) // 2]` no?
If so, then you don't need lines 113 to 115